### DISCUSSION TITLE
How do I omit field from Input type?

### DISCUSSION THREAD

**Discussion by bombillazo** [MichalLytek/type-graphql#1301] [Q&A] [answered]
Hello! If I am using this setup:

```
@ObjectType()
@InputType("PersonInput")
export class Person {

  @Field()
  myField: string;

  @Field()
  optionalField: string;

}
```

How can I omit `optionalField`  from the generated input type?

**Comment by MichalLytek** [comment, marked as answer]
The recommended approach for building similar but different types is mixin-classes:
https://github.com/MichalLytek/type-graphql/tree/master/examples/mixin-classes
